src/algebraic_geometry/projective_spectrum/scheme.lean

```diff
@@ -120,36 +120,245 @@ def degree_zero_part {f : A} {m : ℕ} (f_deg : f ∈ 𝒜 m) : subring (away f)
-  degree_zero_part.num f_deg x ∈ 𝒜 (m * degree_zero_part.deg f_deg x) :=
+lemma degree_zero_part.num_mem {f : A} {m : ℕ} {f_deg : f ∈ 𝒜 m} (x : A⁰_ f_deg) :
+  degree_zero_part.num x ∈ 𝒜 (m * degree_zero_part.deg x) :=
 x.2.some_spec.some.2
 
-lemma degree_zero_part.eq {f : A} {m : ℕ} (f_deg : f ∈ 𝒜 m) (x : A⁰_ f_deg) :
-  x.1 = mk (degree_zero_part.num f_deg x) ⟨f^(degree_zero_part.deg f_deg x), ⟨_, rfl⟩⟩ :=
+lemma degree_zero_part.eq {f : A} {m : ℕ} {f_deg : f ∈ 𝒜 m} (x : A⁰_ f_deg) :
+  x.1 = mk (degree_zero_part.num x) ⟨f^(degree_zero_part.deg x), ⟨_, rfl⟩⟩ :=
 x.2.some_spec.some_spec
 
 lemma degree_zero_part.mul_val {f : A} {m : ℕ} (f_deg : f ∈ 𝒜 m) (x y : A⁰_ f_deg) :
   (x * y).1 = x.1 * y.1 := rfl
 
 end
 
+namespace Proj_iso_Spec_Top_component
+
+/-
+This section is to construct the homeomorphism between `Proj` restricted at basic open set at
+a homogeneous element `x` and `Spec A⁰ₓ` where `A⁰ₓ` is the degree zero part of the localized
+ring `Aₓ`.
+-/
+
+namespace to_Spec
+
+open ideal
+
+-- This section is to construct the forward direction :
+-- So for any `x` in `Proj| (pbo f)`, we need some point in `Spec A⁰_f`, i.e. a prime ideal,
+-- and we need this correspondence to be continuous in their Zariski topology.
+
+variables {𝒜} {f : A} {m : ℕ} (f_deg : f ∈ 𝒜 m) (x : Proj| (pbo f))
+
+/--For any `x` in `Proj| (pbo f)`, the corresponding ideal in `Spec A⁰_f`. This fact that this ideal
+is prime is proven in `Top_component.forward.to_fun`-/
+def carrier : ideal (A⁰_ f_deg) :=
+ideal.comap (algebra_map (A⁰_ f_deg) (away f))
+  (ideal.span $ algebra_map A (away f) '' x.1.as_homogeneous_ideal)
+
+lemma mem_carrier_iff (z : A⁰_ f_deg) :
+  z ∈ carrier f_deg x ↔
+  z.1 ∈ ideal.span (algebra_map A (away f) '' x.1.as_homogeneous_ideal) :=
+iff.rfl
+
+lemma mem_carrier.clear_denominator [decidable_eq (away f)]
+  {z : A⁰_ f_deg} (hz : z ∈ carrier f_deg x) :
+  ∃ (c : algebra_map A (away f) '' x.1.as_homogeneous_ideal →₀ away f)
+    (N : ℕ)
+    (acd : Π y ∈ c.support.image c, A),
+    f ^ N • z.1 =
+    algebra_map A (away f) (∑ i in c.support.attach,
+      acd (c i) (finset.mem_image.mpr ⟨i, ⟨i.2, rfl⟩⟩) * classical.some i.1.2) :=
+begin
+  erw [mem_carrier_iff, ←submodule_span_eq, finsupp.span_eq_range_total, set.mem_range] at hz,
+  rcases hz with ⟨c, eq1⟩,
+  rw [finsupp.total_apply, finsupp.sum] at eq1,
+  obtain ⟨⟨_, N, rfl⟩, hN⟩ := is_localization.exist_integer_multiples_of_finset (submonoid.powers f)
+    (c.support.image c),
+  choose acd hacd using hN,
+  have prop1 : ∀ i, i ∈ c.support → c i ∈ finset.image c c.support,
+  { intros i hi, rw finset.mem_image, refine ⟨_, hi, rfl⟩, },
+
+  refine ⟨c, N, acd, _⟩,
+  rw [← eq1, smul_sum, map_sum, ← sum_attach],
+  congr' 1,
+  ext i,
+  rw [_root_.map_mul, hacd, (classical.some_spec i.1.2).2, smul_eq_mul, smul_mul_assoc],
+  refl
+end
+
+lemma carrier_ne_top :
+  ((x.1.as_homogeneous_ideal.to_ideal : set A) ∩ (submonoid.powers f : set A)) = ∅ →
```

Comment:
```suggestion
  (x.1.as_homogeneous_ideal.to_ideal ∩ submonoid.powers f : set A) = ∅ →
```

Also, would it make sense to move it below `no_intersection` and remove the premise?